elephant/spade.py

```diff
@@ -441,13 +441,15 @@ def concepts_mining(data, binsize, winlen, min_spikes=2, min_occ=2,
     # objects (window positions) and attributes (spikes,
     # indexed with a number equal to  neuron idx*winlen+bin idx)
     context, transactions, rel_matrix = _build_context(binary_matrix, winlen)
-    # By default, set the maximum pattern size to the number of spiketrains
+    # By default, set the maximum pattern size to the maximum number of
+    # spikes in a window
     if max_spikes is None:
-        max_spikes = np.max(np.sum(binary_matrix, axis=-1)) + 1
-    # By default set maximum number of data to number of bins
+        max_spikes = int(np.max(np.sum(rel_matrix, axis=1)))
+    # By default, set maximum number of occurrences to number of non-empty
+    # windows
     if max_occ is None:
-        max_occ = np.max(np.sum(binary_matrix, axis=1)) + 1
-    # check if fim.so available and use it
+        max_occ = (np.sum(np.sum(rel_matrix, axis=1)>0))
```

Comment:
I think you forgot an `int(...)` here or `dtype=int`.

Reply:
It should work nevertheless (also because the test works), but you are right, thanks. Should I fix it or you can do it?

Reply:
I edited it now. Yes, it should work nevertheless. But like this it is more consistent to lines 446-447: 
```python
 max_spikes = int(np.max(np.sum(rel_matrix, axis=1)))
```